### PR TITLE
Create CI/CD pipeline

### PR DESCRIPTION
**Pull request by Kralizek**
This PR adds support for a CI/CD pipeline based on
- [Cake](https://cakebuild.net/)
- [AppVeyor](https://www.appveyor.com/)
- [GitVersion](https://gitversion.net/)
- [JetBrains dotCover](https://www.jetbrains.com/dotcover/)
- [ReportGenerator](https://danielpalme.github.io/ReportGenerator/)

### CODE REVIEW

**Comment by Kralizek** [comment]
@McDoit please check the cake! :)

File: GitVersion.yml [hunk 1]
```diff
@@ -3,5 +3,7 @@ mode: ContinuousDeployment
 continuous-delivery-fallback-tag: preview
 
 branches:
+  master:
+    regex: ^master$|^main$
```
Comment (Kralizek) [inline GitVersion.yml:7]: needed to amend the default configuration of GitVersion so to support `main` as new master branch.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+cache:
+  - '%LocalAppData%\NuGet\v3-cache'
+
+nuget:
+  account_feed: false
+  project_feed: false
+  disable_publish_on_pr: true
+
+install:
+  - cmd: dotnet tool restore
```
Comment (Kralizek) [inline appveyor.yml:14]: now we use the `.config/dotnet-tools.json` to specify which tools need to be installed locally.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - provider: Environment
+    name: NuGet EMG
+    on:
+      branch: master
+      appveyor_repo_tag: true
+  - provider: Environment
+    name: EMG GPR
+    on:
+      branch: master
+      appveyor_repo_tag: true
```
Comment (Kralizek) [inline appveyor.yml]: Support for publishing the package to GitHub Package Registry, only when we apply a tag on master

File: build.cake [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+    .IsDependentOn("Restore")
+    .Does<BuildState>(state =>
+{
+    var settings = new DotNetCorePackSettings
+    {
+        Configuration = "Release",
+        NoRestore = true,
+        OutputDirectory = state.Paths.OutputFolder,
+        MSBuildSettings = new DotNetCoreMSBuildSettings()
+                            .SetInformationalVersion(state.Version.AssemblyVersion)
```
Comment (Kralizek) [inline build.cake:227]: Used to support version output from the tool itself using the `--version` option

File: build.cake [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+    var settings = new DotNetCoreBuildSettings
+    {
+        Configuration = "Debug",
+        NoRestore = true,
+        MSBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("TreatWarningsAsErrors", "True")
+    };
+
+    DotNetCoreBuild(state.Paths.SolutionFile.ToString(), settings);
+});
```
Comment (Kralizek) [inline build.cake:68]: Added to force the static analysis before we go through testing

File: src/Directory.Build.props [hunk 6]
```diff
@@ -5,10 +5,9 @@
   </ItemGroup>
 
   <PropertyGroup>
-    <Configurations>$(Configurations);Verify</Configurations>
     <SkipDefaultEditorConfigAsAdditionalFile>true</SkipDefaultEditorConfigAsAdditionalFile>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <TreatWarningsAsErrors  Condition=" '$(Configuration)'=='Verify' OR '$(Configuration)'=='Release'">true</TreatWarningsAsErrors>
+    <TreatWarningsAsErrors  Condition=" '$(Configuration)'=='Release'">true</TreatWarningsAsErrors>
```
Comment (Kralizek) [inline src/Directory.Build.props:10]: _Warning as errors_ is on by default when building in `Release` mode

File: appveyor.yml (hunk 3 above)
Comment (McDoit) [reply appveyor.yml]: What about "main"?

Comment (Kralizek) [reply appveyor.yml]: Fixed now :)

**Review by McDoit — APPROVED** [review]
LGTM

**Pull request merged**